### DISCUSSION TITLE
3D concave hull from point cloud?

### DISCUSSION THREAD

**Discussion by RyanConway91** [marcomusy/vedo#718] [Q&A]
I have a set of 3D points, each with an associated integer value representing the ID of a given volume (x,y,z,vol_id). Can I use vedo to render the point clouds into the unique volumes? The volumes should have topology. That is, the union of the volumes should completely fill the 3D extent of the points and no volumes should overlap (my points represent a 3D bedrock model). I am thinking what I want is a concave hull of the points within each volume?

**Comment by marcomusy** [comment]
if you need to just visualize the volumes you can split the points by vol_id (with numpy) and shift the Volumes by correct amount. E.g.
```python
from vedo import *

vol1 = Volume(dataurl+"embryo.tif").mode(1)
pts1 = np.random.rand(100,3)*3000 +[6000,3000,3000]
pts1 = Points(pts1)

print(vol1.bounds())

vol2 = Volume(dataurl+"embryo.tif").mode(1).cmap("jet")
vol2.shift(12900,0,0)
pts2 = np.random.rand(100,3)*3000 +[12900+6000,3000,3000]
pts2 = Points(pts2)

show(vol1, pts1, vol2, pts2, axes=1)
```

![Screenshot from 2022-10-26 16-53-46](https://user-images.githubusercontent.com/32848391/198061212-5fadb6d6-e551-447a-9df0-a529549a0d16.png)

**Reply by RyanConway91** [reply]
I need the actual subset volumes for further analysis
